ipapython/install/common.py

```diff
@@ -89,13 +89,24 @@ class Interactive(core.Configurable):
 
 
 class Continuous(core.Configurable):
+    def __init__(self, **kwargs):
+        super(Continuous, self).__init__(**kwargs)
+        self.exceptions_handled = 0
+
     def _handle_execute_exception(self, exc_info):
         try:
             super(Continuous, self)._handle_execute_exception(exc_info)
+        except SystemExit:
+            pass
         except BaseException as e:
             logger.debug("%s", traceback.format_exc())
             if isinstance(e, Exception):
                 logger.error("%s", e)
+            self.exceptions_handled = 1
+
+    def run(self):
+        super(Continuous, self).run()
+        return self.exceptions_handled
 
 
 def installer(cls):
```

Comment:
Why is uninstaller a subclass of the `Continuous` class anyway?

Reply:
Yeah, exactly. Who knows. I can find no design discussion on the installer refactoring, the git commits are terse at best and there is near zero inline documentation.

My working assumption, as mentioned, is that it is because Continuous eats exceptions and expects the uninstaller to keep trying to uninstall stuff. It doesn't seem to do that in practice but that is outside the scope of what I was trying to fix.